### PR TITLE
add schema and migrations for push queue

### PR DESCRIPTION
**Pull request by adityathebe**

### CODE REVIEW

File: models/push_queue.go [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+	"github.com/google/uuid"
+)
+
+// PushQueue represents the push_queue database table.
+type PushQueue struct {
+	ID        uuid.UUID `gorm:"column:id"`
+	ItemID    uuid.UUID `gorm:"column:item_id"`
+	Table     string    `gorm:"column:table_name"`
+	Author    string    `gorm:"column:author"`
```
Comment (moshloop) [inline models/push_queue.go:14]: I don't think this is necessary

File: views/012_changelog.sql [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+      INSERT INTO push_queue (table_name, operation, item_id) VALUES (TG_RELNAME, TG_OP, NEW.component_id);
+      RETURN NEW;
+    ELSIF TG_OP = 'DELETE' THEN
+      INSERT INTO push_queue (table_name, operation, item_id) VALUES (TG_RELNAME, TG_OP, OLD.component_id);
+      RETURN OLD;
+    END IF;
+  
+  ELSIF TG_TABLE_NAME = 'config_relationships' THEN
+    IF TG_OP = 'INSERT' OR TG_OP = 'UPDATE' THEN
+      INSERT INTO push_queue (table_name, operation, item_id) VALUES (TG_RELNAME, TG_OP, NEW.config_id);
```
Comment (moshloop) [inline views/012_changelog.sql:16]: The primary key would be: https://github.com/flanksource/duty/blob/main/schema/config.hcl#L319

File: views/012_changelog.sql [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+      RETURN NEW;
+    ELSIF TG_OP = 'DELETE' THEN
+      INSERT INTO push_queue (table_name, operation, item_id) VALUES (TG_RELNAME, TG_OP, OLD.config_id);
+      RETURN OLD;
+    END IF;
+
+  ELSIF TG_TABLE_NAME = 'check_statuses' THEN
+    IF TG_OP = 'INSERT' OR TG_OP = 'UPDATE' THEN
+      INSERT INTO push_queue (table_name, operation, item_id) VALUES (TG_RELNAME, TG_OP, NEW.check_id);
```
Comment (moshloop) [inline views/012_changelog.sql:25]: Check_id will not be unique, it is check_id + time: https://github.com/flanksource/duty/blob/main/schema/checks.hcl#L103

File: views/012_changelog.sql [hunk 4]
```diff
@@ -0,0 +1,75 @@
+-- Trigger def
+CREATE OR REPLACE FUNCTION change_trigger() 
+RETURNS trigger AS $$
+BEGIN
+  IF TG_TABLE_NAME = 'component_relationships' OR TG_TABLE_NAME = 'config_component_relationships' THEN
```
Comment (moshloop) [inline views/012_changelog.sql:5]: Primary key would be: https://github.com/flanksource/duty/blob/main/schema/components.hcl#L105

File: views/012_changelog.sql (hunk 2 above)
Comment (adityathebe) [reply views/012_changelog.sql:16]: yes. modifying the ones with composite primary keys was on my agenda.

File: views/012_changelog.sql [hunk 5]
```diff
@@ -0,0 +1,94 @@
+-- Trigger def
+CREATE OR REPLACE FUNCTION change_trigger()
```
Comment (yashmehrotra) [inline views/012_changelog.sql:2]: Rename to something more descriptive like `push_changelog_to_event_queue`

Comment (yashmehrotra) [reply views/012_changelog.sql:2]: and rename filename as well

**Pull request merged**